Ebusd compilation fails: src/lib/utils/libutils.so: undefined reference to `SSL_alert_desc_string_long' (and more)

### Description

Since I finally got an Ebusd v5 Adapter, I wanted to install Ebusd on Gentoo Linux. As no packages exists, I started to write my own ebuild. Unfortunately, the compilation itself seems to have a problem with SSL.

Note: While I am using version 23.3, I also applied https://github.com/john30/ebusd/commit/3384f3780087bd6b94d46bf18cdad18201ad516c, as it would fail otherwise due gcc13.

### Actual behavior

```
[34/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -fPIC -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11  -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now -shared -Wl,-soname,libknx.so -o src/lib/knx/libknx.so src/lib/knx/CMakeFiles/knx.dir/knx.cpp.o   && :
[35/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -o src/ebusd/ebusd  -Wl,-rpath,/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/knx:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus/contrib:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/utils:  src/lib/ebus/libebus.so  -lpthread  -lrt  src/lib/knx/libknx.so  -lssl  -lcrypto  src/lib/ebus/contrib/libebuscontrib.so  src/lib/utils/libutils.so && :
FAILED: src/ebusd/ebusd
: && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -o src/ebusd/ebusd  -Wl,-rpath,/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/knx:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus/contrib:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/utils:  src/lib/ebus/libebus.so  -lpthread  -lrt  src/lib/knx/libknx.so  -lssl  -lcrypto  src/lib/ebus/contrib/libebuscontrib.so  src/lib/utils/libutils.so && :
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_alert_desc_string_long'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_free'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_NAME_get_text_by_NID'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_info_callback'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_state_string_long'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_new_connect'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_read_ex'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_callback_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_load_verify_locations'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_get_subject_name'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_free'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get_verify_result'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_write_ex'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_free_all'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_test_flags'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `OPENSSL_init_ssl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_verify'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_new'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `TLS_method'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_new_ssl_connect'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `ERR_reason_error_string'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_default_verify_paths'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_options'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get1_peer_certificate'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_int_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get_state'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `ERR_get_error'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Expected behavior

Compilation should succeed :-)

### ebusd version

23.3

### ebusd arguments

N/A

### Operating system

other

### CPU architecture

x64

### Dockerized

None

### Hardware interface

adapter 5 via USB

### Related integration

other

### Logs

Full Log:

```
>>> Emerging (1 of 1) app-misc/ebusd-23.3::galactica
 * ebusd-23.3.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                               [ ok ]
>>> Unpacking source...
>>> Unpacking ebusd-23.3.tar.gz to /var/tmp/portage/app-misc/ebusd-23.3/work
>>> Source unpacked in /var/tmp/portage/app-misc/ebusd-23.3/work
>>> Preparing source in /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build"
 * Applying ebusd-23.3-gcc13.patch ...                                                                                                                         [ ok ]
 * Hardcoded definition(s) removed in CMakeLists.txt:
 *    set(CMAKE_INSTALL_PREFIX "" CACHE PATH "..." FORCE)
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build"
cmake -C /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/gentoo_toolchain.cmake /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3
loading initial cache file /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
fatal: not a git repository (or any parent up to mount point /var/tmp)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for syslog.h
-- Looking for syslog.h - found
-- Looking for time.h
-- Looking for time.h - found
-- Looking for termios.h
-- Looking for termios.h - found
-- Looking for cfsetspeed
-- Looking for cfsetspeed - found
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found
-- Looking for pselect
-- Looking for pselect - found
-- Looking for ppoll
-- Looking for ppoll - found
-- Looking for timegm
-- Looking for timegm - found
-- Looking for linux/serial.h
-- Looking for linux/serial.h - found
-- Looking for dev/usb/uftdiio.h
-- Looking for dev/usb/uftdiio.h - not found
-- contrib enabled
-- KNX enabled
-- SSL enabled
-- Performing Test HAVE_DIRECT_FLOAT_FORMAT
-- Performing Test HAVE_DIRECT_FLOAT_FORMAT - Success
-- <<< Gentoo configuration >>>
Build type      RelWithDebInfo
Install path    /usr
Compiler flags:
C               -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe
C++             -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11
Linker flags:
Executable      -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now
Module          -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now
Shared          -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now

-- Configuring done (2.6s)
-- Generating done (0.0s)
-- Build files have been written to: /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build
>>> Source configured.
>>> Compiling source in /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build"
ninja -v -j1 -l0
[1/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/bushandler.cpp
[2/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/datahandler.cpp
[3/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/request.cpp
[4/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/network.cpp
[5/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/mainloop.cpp
[6/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/scan.cpp
[7/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/main.cpp
[8/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/main_args.cpp
[9/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/knx -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/ebus -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/../lib/utils  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -MF src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o.d -o src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/ebusd/knxhandler.cpp
[10/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/arg.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/arg.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/arg.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/arg.cpp
[11/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/log.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/log.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/log.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/log.cpp
[12/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/tcpsocket.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/tcpsocket.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/tcpsocket.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/tcpsocket.cpp
[13/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/thread.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/thread.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/thread.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/thread.cpp
[14/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/clock.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/clock.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/clock.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/clock.cpp
[15/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/rotatefile.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/rotatefile.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/rotatefile.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/rotatefile.cpp
[16/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dutils_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -MD -MT src/lib/utils/CMakeFiles/utils.dir/httpclient.cpp.o -MF src/lib/utils/CMakeFiles/utils.dir/httpclient.cpp.o.d -o src/lib/utils/CMakeFiles/utils.dir/httpclient.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/utils/httpclient.cpp
[17/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -fPIC -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11  -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now -shared -Wl,-soname,libutils.so -o src/lib/utils/libutils.so src/lib/utils/CMakeFiles/utils.dir/arg.cpp.o src/lib/utils/CMakeFiles/utils.dir/log.cpp.o src/lib/utils/CMakeFiles/utils.dir/tcpsocket.cpp.o src/lib/utils/CMakeFiles/utils.dir/thread.cpp.o src/lib/utils/CMakeFiles/utils.dir/clock.cpp.o src/lib/utils/CMakeFiles/utils.dir/rotatefile.cpp.o src/lib/utils/CMakeFiles/utils.dir/httpclient.cpp.o   && :
[18/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/result.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/result.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/result.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/result.cpp
[19/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/symbol.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/symbol.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/symbol.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/symbol.cpp
[20/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/filereader.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/filereader.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/filereader.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/filereader.cpp
[21/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/datatype.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/datatype.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/datatype.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/datatype.cpp
[22/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/data.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/data.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/data.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/data.cpp
[23/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/device.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/device.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/device.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/device.cpp
[24/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/transport.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/transport.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/transport.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/transport.cpp
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/transport.cpp: In member function ‘virtual ebusd::result_t ebusd::FileTransport::write(const uint8_t*, size_t)’:
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/transport.cpp:133:36: warning: comparison of integer expressions of different signedness: ‘ssize_t’ {aka ‘long int’} and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  133 |   return (::write(m_fd, data, len) == len) ? RESULT_OK : RESULT_ERR_DEVICE;
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
[25/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/protocol.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/protocol.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/protocol.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/protocol.cpp
[26/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/protocol_direct.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/protocol_direct.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/protocol_direct.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/protocol_direct.cpp
[27/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/message.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/message.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/message.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/message.cpp
[28/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debus_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/ebus/CMakeFiles/ebus.dir/stringhelper.cpp.o -MF src/lib/ebus/CMakeFiles/ebus.dir/stringhelper.cpp.o.d -o src/lib/ebus/CMakeFiles/ebus.dir/stringhelper.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/stringhelper.cpp
[29/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -fPIC -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11  -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now -shared -Wl,-soname,libebus.so -o src/lib/ebus/libebus.so src/lib/ebus/CMakeFiles/ebus.dir/result.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/symbol.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/filereader.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/datatype.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/data.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/device.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/transport.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/protocol.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/protocol_direct.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/message.cpp.o src/lib/ebus/CMakeFiles/ebus.dir/stringhelper.cpp.o  -Wl,-rpath,/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/utils  src/lib/utils/libutils.so && :
[30/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debuscontrib_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/contrib/..  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -Wno-unused-parameter -MD -MT src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/contrib.cpp.o -MF src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/contrib.cpp.o.d -o src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/contrib.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/contrib/contrib.cpp
[31/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Debuscontrib_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/contrib/..  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -Wno-unused-parameter -MD -MT src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/tem.cpp.o -MF src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/tem.cpp.o.d -o src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/tem.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/ebus/contrib/tem.cpp
[32/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -fPIC -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11  -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now -shared -Wl,-soname,libebuscontrib.so -o src/lib/ebus/contrib/libebuscontrib.so src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/contrib.cpp.o src/lib/ebus/contrib/CMakeFiles/ebuscontrib.dir/tem.cpp.o   && :
[33/40] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -D_GNU_SOURCE -Dknx_EXPORTS -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build -I/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src  -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -std=gnu++11 -fPIC   -fpic -Wall -Wno-unused-function -Wextra -Wconversion -Wno-unused-parameter -MD -MT src/lib/knx/CMakeFiles/knx.dir/knx.cpp.o -MF src/lib/knx/CMakeFiles/knx.dir/knx.cpp.o.d -o src/lib/knx/CMakeFiles/knx.dir/knx.cpp.o -c /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knx.cpp
In file included from /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knx.cpp:30:
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h: In member function ‘virtual ebusd::knx_transfer_t ebusd::KnxNetConnection::getPollData(int, uint8_t*, int*, ebusd::knx_addr_t*, ebusd::knx_addr_t*)’:
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h:606:31: warning: conversion from ‘long int’ to ‘int’ may change value [-Wconversion]
  606 |     *recvlen = size > dataLen ? dataLen : size;
      |                ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
In file included from /usr/include/arpa/inet.h:22,
                 from /var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h:22:
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h: In member function ‘const char* ebusd::KnxNetConnection::send(ebusd::knx_transfer_t, ebusd::knx_addr_t, int, const uint8_t*)’:
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h:647:32: warning: conversion from ‘size_t’ {aka ‘long unsigned int’} to ‘__uint16_t’ {aka ‘short unsigned int’} may change value [-Wconversion]
  647 |     h->totalLength.raw = htons(totalLen);
      |                                ^~~~~~~~
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h:664:31: warning: conversion from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Wconversion]
  664 |     l->informationLength = len-1;  // subtracting the TPCI/APCI
      |                            ~~~^~
/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3/src/lib/knx/knxnet.h:675:12: warning: conversion from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Wconversion]
  675 |     d[0] = tpci;
      |            ^~~~
[34/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -fPIC -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11  -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now -shared -Wl,-soname,libknx.so -o src/lib/knx/libknx.so src/lib/knx/CMakeFiles/knx.dir/knx.cpp.o   && :
[35/40] : && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -o src/ebusd/ebusd  -Wl,-rpath,/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/knx:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus/contrib:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/utils:  src/lib/ebus/libebus.so  -lpthread  -lrt  src/lib/knx/libknx.so  -lssl  -lcrypto  src/lib/ebus/contrib/libebuscontrib.so  src/lib/utils/libutils.so && :
FAILED: src/ebusd/ebusd
: && /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -march=native -mfpmath=sse,387 -mtune=intel -O3 -frecord-gcc-switches -fomit-frame-pointer -malign-data=abi -mtls-dialect=gnu2 -pipe -fvisibility-inlines-hidden -std=c++11 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,-O1 -Wl,--sort-common -Wl,-z,now src/ebusd/CMakeFiles/ebusd.dir/bushandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/datahandler.cpp.o src/ebusd/CMakeFiles/ebusd.dir/request.cpp.o src/ebusd/CMakeFiles/ebusd.dir/network.cpp.o src/ebusd/CMakeFiles/ebusd.dir/mainloop.cpp.o src/ebusd/CMakeFiles/ebusd.dir/scan.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main.cpp.o src/ebusd/CMakeFiles/ebusd.dir/main_args.cpp.o src/ebusd/CMakeFiles/ebusd.dir/knxhandler.cpp.o -o src/ebusd/ebusd  -Wl,-rpath,/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/knx:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/ebus/contrib:/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build/src/lib/utils:  src/lib/ebus/libebus.so  -lpthread  -lrt  src/lib/knx/libknx.so  -lssl  -lcrypto  src/lib/ebus/contrib/libebuscontrib.so  src/lib/utils/libutils.so && :
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_alert_desc_string_long'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_free'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_NAME_get_text_by_NID'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_info_callback'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_state_string_long'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_new_connect'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_read_ex'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_callback_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_load_verify_locations'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_get_subject_name'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `X509_free'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get_verify_result'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_write_ex'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_free_all'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_test_flags'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `OPENSSL_init_ssl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_verify'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_new'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `TLS_method'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_new_ssl_connect'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `ERR_reason_error_string'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_default_verify_paths'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_CTX_set_options'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get1_peer_certificate'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `BIO_int_ctrl'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `SSL_get_state'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: src/lib/utils/libutils.so: undefined reference to `ERR_get_error'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
 * ERROR: app-misc/ebusd-23.3::galactica failed (compile phase):
 *   ninja -v -j1 -l0 failed
 *
 * Call stack:
 *     ebuild.sh, line  136:  Called src_compile
 *   environment, line 2191:  Called cmake_src_compile
 *   environment, line  902:  Called cmake_build
 *   environment, line  869:  Called eninja
 *   environment, line 1343:  Called die
 * The specific snippet of code:
 *       "$@" || die -n "${*} failed"
 *
 * If you need support, post the output of `emerge --info '=app-misc/ebusd-23.3::galactica'`,
 * the complete build log and the output of `emerge -pqv '=app-misc/ebusd-23.3::galactica'`.
 * The complete build log is located at '/var/tmp/portage/app-misc/ebusd-23.3/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-misc/ebusd-23.3/temp/environment'.
 * Working directory: '/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3_build'
 * S: '/var/tmp/portage/app-misc/ebusd-23.3/work/ebusd-23.3'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,8 @@ endif()
 
 add_definitions(-fpic -Wall -Wno-unused-function -Wextra)
 
+set(BUILD_SHARED_LIBS OFF)
+
 check_include_file(arpa/inet.h HAVE_ARPA_INET_H)
 check_include_file(dirent.h HAVE_DIRENT_H)
 check_include_file(fcntl.h HAVE_FCNTL_H)

